azure-pipelines.yml

```diff
@@ -0,0 +1,43 @@
+trigger:
+  branches:
+    include:
+    - master
+  tags:
+    include:
+    - v*
+
+strategy:
+  matrix:
+    linux:
+      imageName: 'ubuntu-latest'
+    mac:
+      imageName: 'macos-latest'
+    windows:
+      imageName: 'windows-latest'
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+
+- task: NodeTool@0
+  inputs:
+    versionSpec: '10.x'
+  displayName: 'Install Node.js'
+
+- bash: |
+    /usr/bin/Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+    echo ">>> Started xvfb"
+  displayName: Start xvfb
+  condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+
+- bash: |
+    echo ">>> Compile vscode-test"
+    yarn && yarn compile
+    echo ">>> Compiled vscode-test"
+    cd sample
+    echo ">>> Run sample integration test"
+    yarn && yarn compile && yarn test
```

Comment:
I have no idea what's the difference between them :grin:   I just copied the example in https://code.visualstudio.com/api/working-with-extensions/continuous-integration#azure-pipelines (after fixing it: https://github.com/microsoft/vscode-docs/pull/4328)

Reply:
They're two different package managers for Node.js. I think they offer similar features and it's a matter of preference.
Looking at it a bit more, they're not entirely compatible. For example, yarn has its own lockfile (`yarn.lock`). One of the tests shows this too:
> info No lockfile found.
> warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.

We should stick with one or the other - the syntax should be similar for npm! (e.g. `npm run compile && npm run test`)